databases/functions/userFollowUser/user-follow-add.js

```diff
@@ -15,16 +17,31 @@ module.exports = async (
 ) => {
   if (followedUsers.length === 0) return;
   // User Follow User
-  let follows_array_return = followedUsers.map((val) => {
-    return {
-      //   generate UUID
-      follow_action_id: uuidv4(),
-      user_id_follower: userId,
-      user_id_followed: val
-    };
-  });
 
-  let returnUserFollowUser = await userFollowUserModel.bulkCreate(follows_array_return, {
+  let follows_array = [];
+  await Promise.all(
+    followedUsers.map(async (val) => {
+      //follow sign user
+      follows_array.push({
+        follow_action_id: uuidv4(),
+        user_id_follower: userId,
+        user_id_followed: val
+      });
+
+      const userFollowed = await UsersFunction.findUserById(User, val);
+      if (userFollowed.is_anonymous === false) {
+        const anonymousUser = await UsersFunction.findAnonymousUserId(User, val);
+        follows_array.push({
+          follow_action_id: uuidv4(),
+          user_id_follower: userId,
+          user_id_followed: anonymousUser?.user_id,
+          is_anonymous: true
+        });
+      }
+    })
+  );
+
+  let returnUserFollowUser = await userFollowUserModel.bulkCreate(follows_array, {
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new code introduces a more complex logic for handling follow actions. It creates an array of follow objects based on the followed users, with separate entries for signed-in and anonymous users. These follow objects are then inserted into the database using the `bulkCreate` function. The logic seems correct, but it's important to ensure that the `findUserById` and `findAnonymousUserId` functions from the `UsersFunction` module return the expected results and handle errors properly. Also, consider adding error handling around the `bulkCreate` call to catch any potential database errors.
```diff
-  let follows_array_return = followedUsers.map((val) => {
-    return {
-      //   generate UUID
-      follow_action_id: uuidv4(),
-      user_id_follower: userId,
-      user_id_followed: val
-    };
-  });
- 
-  let returnUserFollowUser = await userFollowUserModel.bulkCreate(follows_array_return, {
-     transaction,
-     returning: true,
-     raw: true
+  let follows_array = [];
+  await Promise.all(
+    followedUsers.map(async (val) => {
+      //follow sign user
+      follows_array.push({
+        follow_action_id: uuidv4(),
+        user_id_follower: userId,
+        user_id_followed: val
+      });
+
+      const userFollowed = await UsersFunction.findUserById(User, val);
+      if (userFollowed.is_anonymous === false) {
+        const anonymousUser = await UsersFunction.findAnonymousUserId(User, val);
+        follows_array.push({
+          follow_action_id: uuidv4(),
+          user_id_follower: userId,
+          user_id_followed: anonymousUser?.user_id,
+          is_anonymous: true
+        });
+      }
+    })
+  );
+
+  let returnUserFollowUser = await userFollowUserModel.bulkCreate(follows_array, {
+     transaction,
+     returning: true,
+     raw: true
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->